openhands-cli/openhands_cli/tui/settings/store.py

```diff
@@ -44,17 +43,38 @@ def load(self) -> Agent | None:
             mcp_config: dict = agent.mcp_config.copy().get('mcpServers', {})
             mcp_config.update(additional_mcp_config)
 
+            # Update LLM metadata with current information
+            agent_llm_metadata = get_llm_metadata(
+                model_name=agent.llm.model,
+                llm_type="agent",
+                session_id=session_id
+            )
+            updated_llm = agent.llm.model_copy(update={"metadata": agent_llm_metadata})
+
+            condenser_updates = {}
+            if agent.condenser and isinstance(agent.condenser, LLMSummarizingCondenser):
+                condenser_updates["llm"] = agent.condenser.llm.model_copy(update={"metadata": get_llm_metadata(
+                    model_name=agent.condenser.llm.model,
+                    llm_type="condenser",
+                    session_id=session_id
+                )})
+
             agent = agent.model_copy(update={
+                "llm": updated_llm,
                 "tools": updated_tools,
                 "mcp_config": {'mcpServers': mcp_config} if mcp_config else {},
-                "agent_context": agent_context
+                "agent_context": agent_context,
+                "condenser": agent.condenser.model_copy(
+                    update=condenser_updates
+                ) if agent.condenser else None
             })
 
             return agent
         except FileNotFoundError:
             return None
-        except Exception:
+        except Exception as e:
             print_formatted_text(HTML("\n<red>Agent configuration file is corrupted!</red>"))
+            raise e
             return None
```

Comment:
We didn't raise and error here so the user could be re-directed into the settings setup flow. The current logic will require the user to manually delete their settings file to use the CLI again